src/main/java/com/google/sps/servlets/ServletUtils.java

```diff
@@ -99,20 +124,21 @@ public static JSONObject readFromApiUrl(URL url) throws IOException {
    *
    * @param datastore the Datastore containing all election data
    * @param electionId the ID of the election being queried
-   * @return the Election Entity if found, null otherwise
+   * @return Optional container that contains either the Election entity's Key or null
    */
   public static Optional<Entity> findElectionInDatastore(
       DatastoreService datastore, String electionId) {
     Query query = new Query("Election");
     PreparedQuery results = datastore.prepare(query);
-    Entity targetElection = null;
+    Entity targetEntity = null;
 
     for (Entity entity : results.asIterable()) {
-      if (entity.getProperty("id") == electionId) {
-        targetElection = entity;
+      if (entity.getProperty("id").equals(electionId)) {
+        targetEntity = entity;
```

Comment:
can you early return here instead (return Optional.of(entity))

And then at the bottom just return Optional.absent()